core/src/main/java/org/axonframework/eventsourcing/eventstore/AbstractEventStorageEngine.java

```diff
@@ -198,8 +201,11 @@ protected void handlePersistenceException(Exception exception, EventMessage<?> f
                                                                            boolean mayBlock);
 
     /**
-     * Returns an optional serialized event entry for given {@code aggregateIdentifier} if the backing database
+     * Returns a stream of serialized event entries for given {@code aggregateIdentifier} if the backing database
      * contains a snapshot of the aggregate.
+     * <p>
+     * It is required that specific event storage engines return snapshots in descending order of their sequence number.
+     * </p>
      *
      * @param aggregateIdentifier The aggregate identifier to fetch a snapshot for
      * @return An optional with a serialized snapshot of the aggregate
```

Comment:
The `@return` documentation is wrong, it states an `Optional` is returned, whilst you're returning a `Stream` now.